src/App.tsx

```diff
@@ -1,24 +1,64 @@
 import React from 'react';
+import { Clock } from './Components/Clock';
+
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean,
+  clockName: number,
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    isClockVisible: true,
+    clockName: 0,
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
```

Comment:
I still can see and click Hide and Set random name buttons when the clock is hidden, even though they are having no effect when clock is hidden, so either hide them or make disabled based on state